File: Codeforces 1349B - Orac and Medians.kt
// https://codeforces.com/problemset/problem/1349/B

import java.io.*
import kotlin.math.max

private var br: BufferedReader? = null

private fun br(): BufferedReader? {
    if (br == null) {
        var isLocal = false
        val file = File("./file/input.txt")

        try {
            isLocal = file.exists()
        } catch (e: Exception) {

        }

        br = if (isLocal) {
            BufferedReader(BufferedReader(FileReader(file)));
        } else {
            BufferedReader(InputStreamReader(System.`in`))
        }
    }
    return br
}

private fun readLn() = br()?.readLine()!! // string line
private fun readInt() = readLn().toInt() // single int
private fun readLong() = readLn().toLong() // single long
private fun readDouble() = readLn().toDouble() // single double
private fun readStrings() = readLn().split(" ") // list of strings
private fun readInts() = readStrings().map { it.toInt() } // list of ints
private fun readLongs() = readStrings().map { it.toLong() } // list of longs
private fun readDoubles() = readStrings().map { it.toDouble() } // list of doubles
private fun readArray() = readStrings().map { it.toInt() }.toIntArray() // list of ints

fun case() {
    var (n, k) = readInts()
    var s = readArray()

    if (n == 1 && s[0] == k) {
        println("yes")
        return
    }

    for (i in 0..n-1) {
        if (s[i] == k) {
            if (i > 0) {
                if (s[i - 1] >= s[i]) {
                    println("yes")
                    return
                }
            }

            if (i < n - 1) {
                if (s[i + 1] >= s[i]) {
                    println("yes")
                    return
                }
            }
        }
    }
    println("no")
}

fun main() {
    var t = readInt()
    while (t-- > 0) {
        case()
    }
}

File: Codeforces 1349B - Orac and Medians.py
# https://codeforces.com/contest/1360/problem/F

import sys
import os
import heapq
import math

try:
    path = "./file/input.txt"
    if os.path.exists(path):
        sys.stdin = open(path, 'r')
    # sys.stdout = open(r"./file/output.txt", 'w')
except:
    pass

t = int(input())

def printd(value):
    # print(value)
    pass

def run(n, k, s):
    i = 0
    isSuccess = True if len(s) == 1 and s[0] == k else False
    while i < n:
        if s[i] != k:
            i += 1
            continue
        
        blow = 0
        biger = 0
        pos = i + 1
        while pos < n:

            if s[pos] >= k:
                biger += 1
            else:
                blow += 1

            if biger - blow == 0 or biger - blow == 1:
                return True

            if s[pos] == k:
                break

            pos += 1

        i = pos
    return isSuccess

def case():
    arr = list(map(int, input().split(" ")))
    (n, k) = arr[0], arr[1]
    s = list(map(int, input().split(" ")))

    isSuccess = run(n, k, s)
    isSuccess = isSuccess or run(n, k, s[::-1])
    
    print("yes" if isSuccess else "no")

for _ in range(t):
    case()